### DISCUSSION TITLE
Feature: check for explicit Qt dependencies

### DISCUSSION THREAD

**Discussion by tlambert03** [chanzuckerberg/napari-hub#333] [Ideas]
From https://github.com/napari/napari/issues/3223#issuecomment-971842237, the culprit is allencell-segmenter, which explicitly declares `pyqt5` [in its dependencies](https://github.com/AllenCell/napari-allencell-segmenter/blob/c732408023c828c07ec2a425f4f426174d94946b/setup.py#L24) which is doubly problematic:

- if the user has installed napari with `conda install napari`, they will have gotten pyqt from conda, (which names it `pyqt`) ... and the pip installation (named `pyqt5`) _will_ cause problems.  They will basically break their environment.
- if the user has pyside2 installed instead of pyqt, they'll may also run into problems.

**Proposal**
plugins should never explicitly require a Qt backend.  The end-user will not have gotten napari running at all if a backend wasn't there, so it's unnecessary, and can only lead to issues.  They should _only_ import from `qtpy` (instead of PyQt5 or PySide2), and not mention a backend.

Would be nice to add a check for this and somehow notify developers.

**Comment by DragaDoncila** [comment]
I absolutely think this is something we should surface for users and plugin developers.

We could surface it only on the preview page initially and give plugin developers time to correct this before displaying it on plugin detail pages.

I do ultimately think that this needs to be shown as a warning to users on individual plugins, because I agree with @tlambert03 that we need to be careful about plugin developer practices reflecting back on napari as instability.

Today we reached over 100 plugins on the napari hub. I'd like us to implement this feature (as well as those in #331 and #334) as a matter of some urgency. This one in particular is relatively trivial. Pinging @neuromusic so we can start roadmapping these.
